app/src/main/java/com/fantastic/bookxchange/activities/NearMeActivity.java

```diff
@@ -0,0 +1,263 @@
+                    try {
+                        addresses = geocoder.getFromLocation(u.getLocation().latitude, u.getLocation().longitude, 1);
+                        Log.d("NearMe", u.getName() + " location: " + addresses.get(0).getLocality() + ", " + addresses.get(0).getAdminArea());
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                    //TODO Intent to User Profile
+
+                    return true;
+                }
+            });
+            populateData();
+        } else {
+            Toast.makeText(this, "Error - Map was null!!", Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+        NearMeActivityPermissionsDispatcher.onRequestPermissionsResult(this, requestCode, grantResults);
+    }
+
+    @NeedsPermission({Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION})
+    void getMyLocation() {
+        //noinspection MissingPermission
+        map.setMyLocationEnabled(true);
+
+        FusedLocationProviderClient locationClient = getFusedLocationProviderClient(this);
+        //noinspection MissingPermission
+        locationClient.getLastLocation()
+                .addOnSuccessListener(new OnSuccessListener<Location>() {
+                    @Override
+                    public void onSuccess(Location location) {
+                        if (location != null) {
+                            onLocationChanged(location);
+                        }
+                    }
+                })
+                .addOnFailureListener(new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+                        Log.d("MapDemoActivity", "Error trying to get last GPS location");
+                        e.printStackTrace();
+                    }
+                });
+    }
+
+    /*
+     * Called when the Activity becomes visible.
+    */
+    @Override
+    protected void onStart() {
+        super.onStart();
+    }
+
+    /*
+     * Called when the Activity is no longer visible.
+     */
+    @Override
+    protected void onStop() {
+        super.onStop();
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+
+        // Display the connection status
+
+        if (mCurrentLocation != null) {
+            LatLng latLng = new LatLng(mCurrentLocation.getLatitude(), mCurrentLocation.getLongitude());
+            CameraUpdate cameraUpdate = CameraUpdateFactory.newLatLngZoom(latLng, 17);
+            map.moveCamera(cameraUpdate);
+            map.animateCamera(CameraUpdateFactory.zoomTo(10), 2000, null);
+
+        } else {
+            Toast.makeText(this, "Current location was null, enable GPS", Toast.LENGTH_SHORT).show();
```

Comment:
you can use `toast("Current location was null, enable GPS");` this method defined in BaseActivity